### ISSUE TITLE
Support defaulting query parameter values when not supplied, and default is set. 

### ISSUE THREAD

**Issue by cdelst** [oapi-codegen/oapi-codegen#1703]
Right now, using the `default` parameter in openAPI specs does functionally nothing in the generator.

We could improve the strict generator by setting the query parameters to the default values set, and not have a pointer to them in the code, marking them as optional.

**Comment by percivalalb** [comment]
An up vote for addtional support for `default` values from me. I came across this when I had a query param that had a default value of `false`, found myself having check if it was nil and assuming a default which introduces a disconnect between the OAPI spec and the code:
```golang
var field bool
if params.Field != nil && params.Field {
    field = params.Field
}

// do something with `field`
```

I think the auto-generated client code requires no changes since a pointer value (!= nil) is used to differentate between a user provided value and no value.

So as you suggest, if the schema has a default value then having that set automatically if the request does not give a value would be a start. Dropping the pointer for a more user-friendly experience would be the next step. In my example I could then just use `params.Field` directly.

I've been having a look at the code changes to support this, I think that passing the default value to https://github.com/oapi-codegen/runtime/blob/912bc9f39f1e8fd175f1f39c48b4adf5ceca39f1/bindparam.go#L309 and using it is the majority of the solution to respecting the `default` value.

**Comment by marcomayer** [comment]
I agree this would be very helpful, right now the default value provided in the spec needs to be manually coded giving way to potential errors.

**Comment by mromaszewicz** [comment]
I agree, it would be very useful. I have a prototype of this working in `oapi-codegen/oapi-codegen-exp`, where every model that we generate has an `ApplyDefaults()` function, which will set default values on unset fields in schemas, however, there's a ptoblem with it, because you can't tell apart the Go zerovalue from a legitimate value set to the zerovalue in cases where the optional/default field isn't referred to by pointer.

That's a long winded way of saying, I don't know how to get it working properly in all cases. It would require having spec-aware JSON unmarshaling.
